src/reducer.js

```diff
@@ -0,0 +1,36 @@
+const initialState = {
+  cats: []
+}
+
+function setState (state, mewState) {
+  return Object.assign({}, state, mewState)
+}
+
+function removeCat (state, id) {
+  return Object.assign({}, state, {
+    cats: state.cats.filter(cat => cat.id !== id)
+  })
+}
+
+function fetchCatsRequest (state) {
+  return Object.assign({}, state)
+}
+
+function fetchCatsSuccess (state, cats) {
+  return Object.assign({}, state, {cats})
+}
+
+export default function rootReducer (state = initialState, action) {
+  switch (action.type) {
+  case "SET_STATE":
+    return setState(state, action.state)
+  case "REMOVE_CAT":
+    return removeCat(state, action.id)
+  case "FETCH_CATS_REQUEST":
+    return fetchCatsRequest(state) // todo okay to have no action?
+  case "FETCH_CATS_SUCCESS":
+    return fetchCatsSuccess(state, action.cats)
```

Comment:
put a fail state - keep a record of the old state and do an optimistic update, then revert on fail. Thats one of the biggest benefits of immutable state